### PR TITLE
Add dbt package to parse the dbt `manifest.json` and `run_results.json`

### LINKED ISSUE #571: Add dbt package to parse dbt artifacts
Add a package to parse the dbt artifacts. Part of resolving #570.

### PR DESCRIPTION
**Pull request by JCZuurmond**
Resolves #571. This PR adds a dbt package to parse the `manifest.json` and the `run_results.json`. 

I will create a separate PR with changes to the core, following the following structure:
1. Create soda cli command to ingest dbt test results.
2. Parses the artifacts
3. Maps them to soda [test results](https://github.com/sodadata/soda-sql/blob/75248e2a4b4bb4b05f3597f5b69f22a5073dad70/core/sodasql/scan/test_result.py#L21)
4. Flushes them to the cloud, like [in the soda scan](https://github.com/sodadata/soda-sql/blob/75248e2a4b4bb4b05f3597f5b69f22a5073dad70/core/sodasql/scan/scan.py#L879)

### CODE REVIEW

**Review by bastienboutonnet — COMMENTED** [review]
I'll leave it to @vijaykiran to review this thoroughly but as far as I can see, I really like the approach and also the fact that you're using dbt's internals to parse and enforce validation which is really nice. 

I think it will also make our life a lot easier if they change the schemas since they seem to provide json schemas we can leverage which is even better.

Might be a good idea to have a quick peek at whether those objects have or are going to change in `v1.0.0` which is in `rc` candidate now.

**Comment by JCZuurmond** [comment]
Thanks @bastienboutonnet . 

note to self and before we merge: add `dbt` to requirements.txt so that it is picked up by the testing suite

File: packages/dbt/sodasql/dbt.py
Comment (vijaykiran) [inline packages/dbt/sodasql/dbt.py:41]: Maybe it's better to assert than raising NotImplementedError

File: packages/dbt/sodasql/dbt.py
Comment (vijaykiran) [inline packages/dbt/sodasql/dbt.py:83]: Same comment as above

**Review by vijaykiran — CHANGES_REQUESTED** [review]
I'm not sure what we are testing in the tests is valuable enough, it looks like we test the python list comprehension. Otherwise, it LGTM!

File: packages/dbt/sodasql/dbt.py
Comment (JCZuurmond) [reply packages/dbt/sodasql/dbt.py:41]: Typically `raise` is used for user facing checks and `assert` for internal checks. The manifest and run_results are provided by the user, so I would keep a `raise`

File: packages/dbt/sodasql/dbt.py
Comment (JCZuurmond) [reply packages/dbt/sodasql/dbt.py:83]: see above

**Comment by JCZuurmond** [comment]
Agreed on the testing, will keep it in for now so that I can use it in a PR for #570

**Review by vijaykiran — APPROVED** [review]

**Pull request merged**
